src/Polly/AsyncPolicy.NonGenericImplementation.cs

```diff
@@ -40,5 +40,6 @@ public abstract partial class AsyncPolicy
         Context context,
         CancellationToken cancellationToken,
         bool continueOnCapturedContext
-    );
+    )
+        where TResult : notnull;
```

Comment:
Does this preclude someone from a return type like `Task<Foo?>`?

Reply:
yes. is that correct?

Reply:
That doesn't sound right to me - if a user wants to return `null` from their policy they should be able to.